Sming/SmingCore/OSTimer.h

```diff
@@ -0,0 +1,101 @@
+/****
+ * Sming Framework Project - Open Source framework for high efficiency native ESP8266 development.
+ * Created 2015 by Skurydin Alexey
+ * http://github.com/anakod/Sming
+ * All files of the Sming Core are provided under the LGPL v3 license.
+ *
+ *
+ * @author: 13 August 2018 - mikee47 <mike@sillyhouse.net>
+ *
+ * This class wraps the OS timer functions.
+ *
+ * For many timing operations the basic OS Timer capabilities are sufficient.
+ * The additional RAM required by the main Timer class soon adds up, so all
+ * short timer requirements are handled by an OSTimer. Still require Timer for
+ * longer periods and the additional callback flexibility.
+ *
+*/
+
+/** @defgroup   timer OSTimer functions
+ *  @brief      Provides timer functions
+*/
+
+#ifndef _SMING_CORE_OSTimer_H_
+#define _SMING_CORE_OSTimer_H_
+
+extern "C" {
+#include "esp_systemapi.h"
+}
+
+/*
+ * According to documentation maximum value of interval for ms
+ * timer after doing system_timer_reinit is 268435ms.
+ * If we do some testing we find that higher values works,
+ * the actual limit seems to be about twice as high
+ * but we use the documented value anyway to be on the safe side.
+*/
+#define MAX_OS_TIMER_INTERVAL_US 268435000
+
+class OSTimer
```

Comment:
Hm... Can we rename this to SimpleTimer ? I am not sure that even 2% of the developers will figure out which timer to prefer for their needs if they have the choice between OSTimer and Timer.